plugins/flashrom/fu-reset-cmos.c

```diff
@@ -0,0 +1,55 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <sys/io.h>
+#include <unistd.h>
+
+/* From coreboot's src/include/pc80/mc146818rtc.h file */
+#define RTC_BASE_PORT 0x70
+
+/*
+ * This is the offset of the first of the two checksum bytes
+ * we may want to figure out how we can determine this dynamically
+ * during execution.
+ */
+#define CMOS_CHECKSUM_OFFSET 123
+
+void
+cmos_checksum_write(uint16_t checksum);
+void
+cmos_write_byte(uint8_t addr, uint8_t val);
+
+void
+fu_reset_cmos(void)
+{
+	/* Call ioperm() to grant us access to ports 0x70 and 0x71 */
+	ioperm(RTC_BASE_PORT, 2, true);
+
+	/* Write a default value to the CMOS checksum */
+	cmos_checksum_write(0xffff);
+}
+
+void
+cmos_checksum_write(uint16_t checksum)
+{
+	unsigned char lo, hi;
+
+	/* The checksum is stored in a big-endian format. */
+	hi = (unsigned char)(checksum >> 8);
+	lo = (unsigned char)(checksum & 0x00ff);
+	cmos_write_byte(CMOS_CHECKSUM_OFFSET, hi);
+	cmos_write_byte(CMOS_CHECKSUM_OFFSET + 1, lo);
+}
+
+void
+cmos_write_byte(uint8_t addr, uint8_t val)
+{
+	/* Reject addresses in the second bank */
+	if (addr >= 128)
+		return;
```

Comment:
is this an error, or just something to ignore?

Reply:
Just something to ignore